src/main/scala/org/allenai/wikitables/WikiTablesSemanticParserCli.scala

```diff
@@ -170,19 +190,31 @@ class WikiTablesSemanticParserCli extends AbstractCli() {
     val sempreEntityLinking = WikiTablesDataProcessor.getEntityLinking(ex)
     val builder = ListBuffer[(Span, Entity, List[Int], Double)]()
     for (linking <- sempreEntityLinking.asScala) {
-      val entityExpr = Expression2.constant(linking.getSecond().toString)
-      val entityType = WikiTablesTypeDeclaration.CELL_TYPE
+      val entityString = linking.getSecond.toString
+      val entityExpr = Expression2.constant(entityString)
+      val entityType = typeDeclaration.getType(entityString)
       val template = ConstantTemplate(entityType, entityExpr)
       // TODO: Passing names as null. Not sure what happens.
       val entity = Entity(entityExpr, entityType, template, null)
```

Comment:
So names is a list of list of ints. I'm assuming each inner list is a sequence of token ids that correspond to the entity?